aten/src/ATen/native/sparse/SparseBinaryOpIntersectionCommon.h

```diff
@@ -262,23 +235,32 @@ void _sparse_binary_op_intersection_kernel_impl(
   // which is implicit in the definition of hash_coeffs,
   // it could be shown that the hash function is actually bijective and, hence,
   // is a perfect hash function (no collisions ever).
-  const auto kHash = std::is_same<hash_t, int64_t>::value ? kLong : kInt;
-  const auto hash_coeffs = [&]() -> Tensor {
+
+  // Need owning storage in case of the Tensor class.
+  const auto hash_coeffs_storage = [&]() -> auto {
     const auto broadcasted_sparse_dim_shape = std::vector<int64_t>(
       broadcasted_shape.begin(),
       broadcasted_shape.begin() + probably_coalesced.sparse_dim()
     );
-    auto strides = contiguous_strides(broadcasted_sparse_dim_shape);
-    auto strides_len = static_cast<int64_t>(strides.size());
-    auto hash_coeffs = at::empty(
-        {strides_len},
-        probably_coalesced._indices().options().device(kCPU).dtype(kHash));
-    // Copy with a potential casting. Is there a nicer way?
-    for (const auto i : c10::irange(strides_len)) {
-      hash_coeffs[i] = strides[i];
+    auto strides = c10::contiguous_strides(broadcasted_sparse_dim_shape);
+
+    if constexpr (max_static_len > 0) {
+      std::array<int64_t, max_static_len> strides_as_array;
+      std::copy(strides.begin(), strides.end(), strides_as_array.begin());
+      return strides_as_array;
+    } else {
+      auto strides_as_tensor = at::tensor(strides, probably_coalesced._indices().options().device(kCPU).dtype(kLong));
```

Comment:
`probably_coalesced._indices().options().dtype()` is already `kLong`:
```suggestion
      auto strides_as_tensor = at::tensor(strides, probably_coalesced._indices().options().device(kCPU));
```

Reply:
Not really in general, it is of type `index_t`. It is set outside to int64, but that could change in the future I assume...